pkg/iter/entry_iterator.go

```diff
@@ -193,18 +193,6 @@ func (i *mergeEntryIterator) Next() bool {
 		return false
 	}
 
-	// shortcut for the last iterator.
```

Comment:
Without tracking previous, this trick doesn't work for a single stream anymore

Reply:
Can you elaborate a bit?

Reply:
Sure! In the case there is only one iterator, this shortcut doesn't store previous entries or check current entries against previous entries in the buffer so it allows any duplicates through. Making the code do that starts to look a lot like what's going on below.

Reply:
I think this is only a problem if there are duplicates within a single iterator. If we avoid that happening, these optimizations still work.